build/Rulesets/AnalyzerProject_BuildRules.ruleset

```diff
@@ -1,29 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<RuleSet Name="Common diagnostic rules for all analyzer projects" Description="Enables/disable rules specific to all analyzer projects." ToolsVersion="14.0">
-  <Include Path=".\Roslyn_BuildRules.ruleset" Action="Default" />
-
-  <Rules AnalyzerId="Microsoft.CodeAnalysis.Analyzers" RuleNamespace="Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers">
```

Comment:
We have analyzer project specific rulesets because these MetaAnalyzers are only useful for projects which contain a diagnostic analyzer/code fix. If we are removing these rulesets because we think it is fine to load and execute these analyzers for all projects, we should also remove [this](https://github.com/dotnet/roslyn/blob/master/build/Rulesets/Roslyn_BuildRules.ruleset#L72) configuration that disabled metadata analyzers for non-analyzer projects.

Reply:
I'm removing this file because it is not used. No project sets `AnalyzerProject` msbuild property, which would cause this file to be used. Should it actually be used?

Reply:
There is also this comment in the ruleset file:
> https://github.com/dotnet/roslyn/issues/26420: Enable rule RS1022 for Roslyn.sln

Reply:
Seems it got broken somewhere after that commit. We still need to retain suppression for RS1022

Reply:
Instead of having special ruleset for projects that declare analyzers, isn't it better to have the applicable rules in the common Shipping ruleset, even if they are no-ops for projects that do not declare analyzers? That way we don't need to worry about manually tracking which project has analyzers and which does not.

Reply:
Yes, that seems fine - we should remove the bulk meta-analyzer suppression [here](https://github.com/dotnet/roslyn/blob/master/build/Rulesets/Roslyn_BuildRules.ruleset#L72) and only retain the suppressions that we really want (for example RS1022 that you mentioned above)

Reply:
https://github.com/dotnet/roslyn/issues/31654

Reply:
Thanks!